### DISCUSSION TITLE
[A-L] (2023/24) Foglio 8 - Esercizio 7

### DISCUSSION THREAD

**Discussion by Elia-Belli** [sapienzastudentsnetwork/calcolo-delle-probabilita#195] [Esercizi - Bertini] [answered]
![Screenshot 2023-11-20 224020](https://github.com/sapienzastudentsnetwork/calcprob2324/assets/107944476/3e7da3d3-a0d5-4b1d-afc5-60260e00ba5d)

**Comment by cornflexxx** [comment]
$1)$ Chiamiamo $S$ la _v.a_ che indica il numero di  prodotti scartati $s$.
_OSS_:
 $$S=\sum_{i=1}^{n}S_i=s$$
Quindi $S_i$ è la $v.a$ che vale 1 quando il prodotto controllato è difettoso (di conseguenza viene scartato) , e vale 0 sia quando il prodotto ispezionato è funzionante  sia quando il prodotto non viene ispezionato. Possiamo riscrivere il precedente ragionamento in questo modo , considerando le probabilità $p$ e $\alpha$ fornite dall'esercizio:
$$\mathbb{P}(S_i=1)=p\alpha$$

$$\mathbb{P}(S_i=0)=[(p-1)\alpha]\cup [p(\alpha -1)]\cup [(1-p)(\alpha-1)]$$


È necessario  riscrivere $S$ alla luce di  queste distribuzioni come quattro variabili aleatorie diverse: $X_s,X_{c-s},X_{d},X_f$; dove nell'ordine in cui sono scritte indicano: il numero di prodotti difettosi controllati, il numero di prodotti  funzionanti controllati e il numero di prodotti non controllati (funzionanti e non). Quindi $S$  ha una distribuzione multinomiale :
$$(X_s,X_{c-s},X_{d},X_f)\sim Multinom(n,\alpha p,\alpha(1-p),p(\alpha -1),(1-p)(\alpha-1))$$

$$\mathbb{P}(S=s)=\mathbb{P}(X_s=s,X_{c-s}=c-s,X_d=d,X_f=n-c-d)=...$$

$$...=\binom{n}{s,c-s,d,n-c-d}(\alpha p)^s\cdot\[\alpha(1-p)]^{c-s}\cdot [p(\alpha -1)]^d\cdot [(1-p)(\alpha-1)]^{n-c-d} $$

$2)$  Il secondo punto risulta nello svolgimento simile al primo , in questo caso però ci troveremo una distribuzione multinomiale a tre $v.a$, infatti   ci interessano solamente i prodotti che hanno passato la fase di controllo e ora sono in vendita. 
All'atto pratico ci basta escludere $X_k$(i prodotti difettosi che sono stati controllati e scartati), e quindi abbiamo:
$$(X_{c-k},X_{d},X_f)\sim Multinom(n-k,\alpha(1-p),p(\alpha -1),(1-p)(\alpha-1))$$
$$\mathbb{P}(D=d)=\mathbb{P}(X_d=d,X_{c-k}=c-k,X_f=n-c-d)=...\\qquad \text{Indico con D la v.a. che conta i prod. difettosi venduti}$$

$$...=\binom{n-k}{d,c-k,n-c-d}[p(\alpha -1)]^d\cdot [\alpha(1-p)]^{c-k}\cdot [(1-p)(\alpha-1)]^{n-c-d} \\quad \text{Con k=0,1...,n}$$

(Nel secondo punto $k=s$)

**Comment by CiottoloMaggico** [comment, marked as answer]
## Soluzione confermata in classe
Per il primo punto procedo come segue:
@cornflexxx credo tu ti sia dimenticato una sommatoria al primo punto, così stai contando la probabilità di un particolare evento dato che il numero di funzionanti controllati, funzionanti non controllati, rotti non controllati e rotti controllati è fissato. 
![image](https://github.com/sapienzastudentsnetwork/calcprob2324/assets/79638284/eb21e902-eb52-4b7c-b2c6-88d75f7cae90)
Questi risultati sono verificati da questo codice:
```
import math
import random

def binomial_prob(n, k, a, p):
    return math.comb(n, k)*(a*p)**k*(1-a*p)**(n-k)

def multinom(n, f1, f2, r1, r2, a, p):
    return (math.factorial(n)/(math.factorial(f1)*math.factorial(f2)*math.factorial(r1)*math.factorial(r2)))*((1-p)*a)**f1*((1-p)*(1-a))**f2*(p*(1-a))**r2*(a*p)**r1

if __name__ == "__main__":
    n = 10
    k = 3
    p = random.uniform(0.1,0.99)
    a = random.uniform(0.1,0.99)
    multinomial = 0


    binomial = binomial_prob(n, k, a, p)

    for i in range(0, n-k+1):
        for j in range(0, n-k-i+1):
                print(i, "\n", j , "\n", n-k-i-j)
                multinomial += multinom(n, i, j, k, n-k-i-j, a, p)

    print(binomial)
    print(multinomial)

```
Per il punto due utilizzerei la distribuzione condizionata, dato che il numero di scartati è conosciuto a priori
![image](https://github.com/sapienzastudentsnetwork/calcprob2324/assets/79638284/7b9be40c-fe4f-411f-ad1f-dfaa4cde0c3b)

**Comment by LeoCosta02** [comment]
@CiottoloMaggico nel punto 2 perchè consideri solo F e non Fc e Fn ?

**Comment by CiottoloMaggico** [comment]
Perchè ci interessa sapere solo i prodotti rotti che non sono stati testati e quindi messi in commercio. Distinguere tra i funzionanti controllati e non non cambia il risultato.

**Comment by LeoCosta02** [comment]
@CiottoloMaggico grazie
